### PR TITLE
BCDA-4213 - Update service.go to add termination date information 

### PR DESCRIPTION
**Pull request by mtrang1263**
### Fixes [BCDA-4213](https://jira.cms.gov/browse/BCDA-4213)

To implement the following tickets:

1. https://jira.cms.gov/browse/BCDA-4214
2. https://jira.cms.gov/browse/BCDA-4216
3. https://jira.cms.gov/browse/BCDA-4217

We need access to the various date parameters related to an ACO's termination.

Our original strategy was to add this as a K:V pair on the context (using `context.WithValue`). However, these data parameters are only needed within service.go and can be populated there. Also, we only need this termination date info in one method `GetQueJobs`

Instead of using contexts and using context.Value(), we introduce a new RequestConditions struct that contains information about the data request. It includes ACO information, job details, timing information, and termination date info. 

### Change Details
1. Introduce RequestConditions struct that captures all information needed to generate job items. Before, we had a function call with several arguments.
2. Update service.go to read in termination details and populate the various dates.

### Security Implications

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
1. Added unit tests to verify that dates are populated correctly.
2. Deploy to dev environment and run smoke tests to ensure the request parameters are set as expected. [Successful Run](https://bcda-ci.adhocteam.us/job/BCDA%20-%20Test%20-%20Integration%20Smoke/2936/)

### CODE REVIEW

File: bcda/api/requests.go [hunk 1]
```diff
@@ -346,7 +346,19 @@ func (h *Handler) bulkRequest(resourceTypes []string, w http.ResponseWriter, r *
...
+		Resources: resourceTypes,
+
+		CMSID: ad.CMSID,
+		ACOID: newJob.ACOID,
+
+		JobID:           newJob.ID,
+		Since:           since,
+		TransactionTime: newJob.TransactionTime,
+	}
+	queJobs, err = h.Svc.GetQueJobs(ctx, conditions)
```
Comment (katyasoup) [inline bcda/api/requests.go:361]: 🎉 love this, so much easier to read

File: bcda/models/service.go [hunk 2]
```diff
@@ -348,6 +381,27 @@ func (s *service) getBenesByFileID(ctx context.Context, cclfFileID uint) ([]*CCL
 	return benes, nil
 }
 
+// setTimeConstraints searches for any time bounds that we should apply on the associated ACO
```
Comment (katyasoup) [inline bcda/models/service.go:384]: nice!

**Review by katyasoup — APPROVED** [review]
LGTM!

File: bcda/api/requests.go (hunk 1 above)
Comment (CosmicSyntax) [reply bcda/api/requests.go:361]: Keyed fields 👍

**Review by CosmicSyntax — APPROVED** [review]
LGTM

**Pull request merged**